database/RecreateOPMDatabase.sql

```diff
@@ -119,3 +119,91 @@ CREATE TABLE Comments (
 );
 
 
+-- FUNCTIONS
+-- Updates the last_updated of Projects table
+CREATE OR REPLACE FUNCTION update_project_last_updated()
+RETURNS TRIGGER AS $$
+BEGIN
+    UPDATE Projects SET last_updated = CURRENT_TIMESTAMP WHERE project_id = NEW.project_id;
+    RETURN NEW;
+END;
+$$ LANGUAGE plpgsql;
+
+-- Updates the last_updated of Tasks table
+CREATE OR REPLACE FUNCTION update_task_last_updated()
+RETURNS TRIGGER AS $$
+BEGIN
+    UPDATE Tasks SET last_updated = CURRENT_TIMESTAMP WHERE task_id = NEW.task_id;
+    RETURN NEW;
+END;
+$$ LANGUAGE plpgsql;
+
+-- Updates last_updated of called for Table
+CREATE OR REPLACE FUNCTION update_last_updated()
+RETURNS TRIGGER AS $$
+BEGIN
+    NEW.last_updated = CURRENT_TIMESTAMP;
+    RETURN NEW;
+END;
+$$ LANGUAGE plpgsql;
```

Comment:
What's the difference between these three? Do we just need the first two? I'm totally out of the loop on trigger functions, not sure what the third one is doing vs the first two.

Reply:
For some reason calling the task specific function for updating the task threw an error, similar with the project function when triggering the project. I think something about how SQL doesn't like it when a table's trigger references a function that is also referencing the same table? Not too sure though
The third one is used for those two cases where task is updating itself and the project is updating itself.